Sources/XMLCoder/Auxiliaries/XMLStackParser.swift

```diff
@@ -126,6 +126,10 @@ extension XMLStackParser: XMLParserDelegate {
                 namespaceURI: String?,
                 qualifiedName: String?,
                 attributes attributeDict: [String: String] = [:]) {
+        #if os(Linux)
+            // For some reason, element names on linux are coming out with the namespace after the name
+            let elementName = elementName.components(separatedBy: ":").reversed().joined(separator: ":")
```

Comment:
this is most probably caused by [this issue](https://bugs.swift.org/browse/SR-10424). I wonder if this breaks any of the tests we have for the namespaces, since `XMLDecoder` has a special boolean flag for namespace handling.

Reply:
All test were passing when I tried it. Do we need additional tests to make sure something else specific isn't broken?

Reply:
sorry, it seems I previously missed a test case [here](https://github.com/MaxDesiatov/XMLCoder/blob/e17d335683faefa556e820ba0423c3493c2d35ec/Tests/XMLCoderTests/NamespaceTest.swift#L194). We need an additional test where namespaces are present in the source XML, but `shouldProcessNamespaces` is `false`. This can be a copy of the existing test function, but with that property flipped and an assert verifying that the namespace part isn't stripped after decoding.

Reply:
another thing in this new namespace test: you'd most probably need to preserve the namespace in coding keys, which probably requires copying the decoded type and adding the namespaces there in the coding keys of the copy. Hope this makes sense, please let me know if there are any questions.